### PR TITLE
Hw-3

### PR DESCRIPTION
**Pull request by Kaijou88**

### CODE REVIEW

File: src/main/java/core/basesyntax/BinaryString.java [hunk 1]
```diff
@@ -7,6 +7,19 @@
      * а возвращает String с представлением этого числа в двоичном виде.
      */
     public String toBinaryString(int value) {
-        return null;
+        StringBuilder binary = new StringBuilder();
+        int remainder;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/BinaryString.java]: let's solve without this variable `remainder`

File: src/main/java/core/basesyntax/BinaryString.java [hunk 2]
```diff
@@ -7,6 +7,19 @@
...
+        if (value == 0 | value == 1) {
+            binary.append(value);
+        } else {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/BinaryString.java]: try to solve without `else` statement

File: src/main/java/core/basesyntax/BinaryString.java (hunk 1 above)
Comment (Kaijou88) [reply src/main/java/core/basesyntax/BinaryString.java]: done

File: src/main/java/core/basesyntax/BinaryString.java (hunk 2 above)
Comment (Kaijou88) [reply src/main/java/core/basesyntax/BinaryString.java]: done

File: src/main/java/core/basesyntax/BinaryString.java [hunk 3]
```diff
@@ -7,6 +7,16 @@
      * а возвращает String с представлением этого числа в двоичном виде.
      */
     public String toBinaryString(int value) {
-        return null;
+        StringBuilder binary = new StringBuilder();
+        if (value == 0 | value == 1) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/BinaryString.java]: try to solve without ` | value == 1`

File: src/main/java/core/basesyntax/BinaryString.java [hunk 4]
```diff
@@ -7,6 +7,16 @@
...
+            return new String(binary.append(value));
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/BinaryString.java]: why do you use StringBuilder here?

File: src/main/java/core/basesyntax/BinaryString.java [hunk 5]
```diff
@@ -7,6 +7,16 @@
...
+        }
+        while (value > 1) {
+            binary.append(value % 2);
+            value = (value - value % 2) / 2;
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/BinaryString.java]: ` - value % 2` - do we need this?

File: src/main/java/core/basesyntax/BinaryString.java [hunk 6]
```diff
@@ -7,6 +7,16 @@
...
+        }
+        binary.append(value);
+        binary.reverse();
+        return new String(binary);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/BinaryString.java]:
```suggestion
        return binary.reverse().toString();
```

File: src/main/java/core/basesyntax/BinaryString.java (hunk 3 above)
Comment (Kaijou88) [reply src/main/java/core/basesyntax/BinaryString.java]: done

File: src/main/java/core/basesyntax/BinaryString.java (hunk 4 above)
Comment (Kaijou88) [reply src/main/java/core/basesyntax/BinaryString.java]: done

File: src/main/java/core/basesyntax/BinaryString.java (hunk 5 above)
Comment (Kaijou88) [reply src/main/java/core/basesyntax/BinaryString.java]: done

File: src/main/java/core/basesyntax/BinaryString.java [hunk 7]
```diff
@@ -7,6 +7,15 @@
      * а возвращает String с представлением этого числа в двоичном виде.
      */
     public String toBinaryString(int value) {
-        return null;
+        if (value == 0) {
+            return String.valueOf(value);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/BinaryString.java]:
```suggestion
            return "0";
```

File: src/main/java/core/basesyntax/BinaryString.java [hunk 8]
```diff
@@ -7,6 +7,15 @@
...
+        }
+        StringBuilder binary = new StringBuilder();
+        while (value > 1) {
+            binary.append(value % 2);
+            value /= 2;
+        }
+        binary.append(value);
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/BinaryString.java]:
```suggestion
        while (value > 0) {
            binary.append(value % 2);
            value /= 2;
        }
```

**Review by Tarabarnik — APPROVED** [review]

**Pull request closed without merging**